test/WebJobs.Script.Tests/AdminControllerTests.cs

```diff
@@ -1,20 +1,94 @@
+using Moq;
 using Xunit;
 
 namespace Microsoft.Azure.WebJobs.Script.Tests
 {
     public class AdminControllerTests
     {
+        private Mock<ScriptHost> hostMock;
+        private Mock<WebScriptHostManager> managerMock;
+        private Collection<FunctionDescriptor> testFunctions;
+        private AdminController testController;
+
+        public AdminControllerTests()
+        {
+            hostMock = new Mock<ScriptHost>(MockBehavior.Strict, new object[] { new ScriptHostConfiguration() });
+            hostMock.Setup(p => p.Functions).Returns(testFunctions);
+
+            managerMock = new Mock<WebScriptHostManager>(MockBehavior.Strict);
+            managerMock.SetupGet(p => p.Instance).Returns(hostMock.Object);
+
+            testController = new AdminController(managerMock.Object);
+        }
+
         [Fact]
-        public void AdminController_HasAuthorizationLevelAttribute()
+        public void HasAuthorizationLevelAttribute()
         {
             AuthorizationLevelAttribute attribute = typeof(AdminController).GetCustomAttribute<AuthorizationLevelAttribute>();
             Assert.Equal(AuthorizationLevel.Admin, attribute.Level);
         }
+
+        [Fact]
+        public async Task Invoke_CallsFunction()
+        {
+            string testFunctionName = "TestFunction";
+            string triggerParameterName = "testTrigger";
+            string testInput = Guid.NewGuid().ToString();
+            bool functionInvoked = false;
+            
+            hostMock.Setup(p => p.CallAsync(It.IsAny<string>(), It.IsAny<Dictionary<string, object>>(), CancellationToken.None))
+                .Callback<string, Dictionary<string, object>, CancellationToken>((name, args, token) =>
+                {
+                    functionInvoked = true;
+
+                    // verify the correct arguments were passed to the invoke
+                    Assert.Equal("Test", name);
+                    Assert.Equal(1, args.Count);
+                    Assert.Equal(testInput, (string)args[triggerParameterName]);
+                })
+                .Returns(Task.CompletedTask);
+
+            // Add a few parameters, with the trigger parameter last
+            // to verify parameter order handling
+            Collection<ParameterDescriptor> parameters = new Collection<ParameterDescriptor>
+            {
+                new ParameterDescriptor("context", typeof(ExecutionContext)),
+                new ParameterDescriptor("log", typeof(TraceWriter)),
+                new ParameterDescriptor(triggerParameterName, typeof(string))
+                {
+                    IsTrigger = true
+                }
+            };
+            testFunctions.Add(new FunctionDescriptor(testFunctionName, null, null, parameters));
+
+            FunctionInvocation invocation = new FunctionInvocation
+            {
+                Input = testInput
+            };
+            HttpResponseMessage response = testController.Invoke(testFunctionName, invocation);
```

Comment:
10 seconds to fix the bug, 45 minutes to put mock testing infrastructure in place for it :)